src/components/feedback/Dialog.tsx

```diff
@@ -73,21 +88,32 @@ export default function Dialog({
   initialFocus = 'auto',
   restoreFocus = false,
   transitionComponent: TransitionComponent,
-  variant = 'panel',
 
   classes,
   elementRef,
-
-  // Forwarded to Panel
-  buttons,
-  icon,
   onClose,
-  paddingSize = 'md',
-  scrollable = true,
-  title,
 
-  ...htmlAttributes
+  ...rest
 }: DialogProps) {
+  const isPanel = isPanelProps(rest);
+  const {
+    buttons,
+    icon,
+    paddingSize = 'md',
+    title = '',
+    scrollable = true,
+    ...htmlAttributes
+  } = isPanel
```

Comment:
An alternative approach to forwarding all the props would be to put all the panel props under a specific key. This would be a little less ergonomic to use in the common case that the dialog is a panel though.